docs/guides/using-org-slugs.mdx

```diff
@@ -0,0 +1,397 @@
+
+  This sample application uses the [personal account](https://clerk.com/glossary#personal-account), so you can set `hidePersonal` to `false` to ensure the personal
+  account is selectable. You can also allow users to customize their organization's URL slug when they
+  create their organization initially by setting `hideSlug` to `false`.
+
+  Finally, you can give the components navigation instructions. If the user selects or creates an organization
+  with the slug `acmecorp`, by setting `afterCreateOrganizationUrl` and `afterSelectOrganizationUrl` to `'/orgs/:slug'`,
+  the component will navigate the user to `/orgs/acmecorp`. If the user selects their [personal account](https://clerk.com/glossary#personal-account),
+  they will be navigated to `/me` via `afterSelectPersonalUrl`.
+
+  <Tabs items={["<OrganizationSwitcher />", "<OrganizationList />"]}>
+    <Tab>
+      ```tsx {{ filename: 'app/sidebar.tsx', mark: [6] }}
+      import { OrganizationSwitcher } from '@clerk/nextjs'
+
+      export default function Sidebar() {
+        return (
+          <OrganizationSwitcher
+            // prettier-ignore
+            hidePersonal={false}
+            hideSlug={false}
+            afterCreateOrganizationUrl="/orgs/:slug"
+            afterSelectOrganizationUrl="/orgs/:slug"
+            afterSelectPersonalUrl="/me"
+          />
+        )
+      }
+      ```
+    </Tab>
+
+    <Tab>
+      ```tsx {{ filename: 'app/organization-list/[[...organization-list]]/page.tsx', mark: [6] }}
+      import { OrganizationList } from '@clerk/nextjs'
+
+      export default function OrganizationListPage() {
+        return (
+          <OrganizationList
+            // prettier-ignore
+            hidePersonal={false}
+            hideSlug={false}
+            afterCreateOrganizationUrl="/orgs/:slug"
+            afterSelectOrganizationUrl="/orgs/:slug"
+            afterSelectPersonalUrl="/me"
+          />
+        )
+      }
+      ```
+    </Tab>
+  </Tabs>
+
+  > [!TIP]
+  > If you prefer to have an [organization ID](/docs/references/javascript/organization/organization#properties) appear in
+  > your URL rather than a slug, you can use `:id` instead of `:slug` with both `afterCreateOrganizationUrl` and
+  > `afterSelectOrganizationUrl`.
+
+  ### Using clerkMiddleware to set the active organization
+
+  With [clerkMiddleware](https://clerk.com/docs/references/nextjs/clerk-middleware), you can
+  declare URL patterns that indicate that a specific organization or the user's [personal account](https://clerk.com/glossary#personal-account)
+  should be activated. If the middleware detects one
+  of these patterns in the URL, and it notices that a different organization is
+  currently active according to the session, it will activate the indicated organization if
+  possible.
+
+  > [!NOTE]
+  > Implementation Detail: To resolve a mismatch, Clerk middleware initiaties a [Handshake](docs/references/sdk/terminology#:~:text=each:Handshake).
```

Comment:
How useful this information would be to developers? What do you think of removing this part?

I think a handshake is an implementation detail that we mostly expose/educate for SDK development: https://clerk.com/docs/pr/1664/references/sdk/fullstack#add-handshake-support

Reply:
No, we don't have docs on the handshake yet. 

I'm in favour of removing this

Reply:
👍  Away it goes!

44b5482fff38ab78f72e46ad80a6e7897af7fd74